homeassistant/components/group.py

```diff
@@ -249,40 +269,75 @@ def assumed_state(self):
 
     def update_tracked_entity_ids(self, entity_ids):
         """Update the member entity IDs."""
-        self.stop()
+        run_callback_threadsafe(
+            self.hass.loop, self.async_update_tracked_entity_ids, entity_ids
+        ).result()
+
+    def async_update_tracked_entity_ids(self, entity_ids):
+        """Update the member entity IDs.
+
+        This method must be run in the event loop.
+        """
+        self.async_stop()
         self.tracking = tuple(ent_id.lower() for ent_id in entity_ids)
         self.group_on, self.group_off = None, None
 
-        self.update_ha_state(True)
-
-        self.start()
+        self.hass.async_run_job(self.async_update_ha_state, True)
+        self.async_start()
```

Comment:
This logic is now slightly different. The `async_run_job` might not be done yet before `async_start` starts. Can this be a problem?